File: src/test/cpp/sorting/heapsort/heap_sort_test.cpp
//
// Created by nisha on 6/2/2020.
//

#include <algorithm>

#include "gtest/gtest.h"
#include "../../../../main/cpp/sorting/heapsort/heap_sort.hpp"

TEST(Sorting, HeapSortTest) {
    for(int i = 0; i < 1000; i ++) {
        std::vector<int> arr;

        for(int j = 0; j < 500; j ++) {
            arr.push_back(rand() % 1000);
        }

        std::vector<int> temp = arr;
        std::sort(temp.begin(), temp.end());
        heap_sort(arr);
        ASSERT_EQ(temp, arr);
    }
}

File: java/src/test/java/sorting/HeapSortTest.java
package sorting;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import sorting.heapsort.HeapSort;

public class HeapSortTest {

  private int[] arr;

  @BeforeEach
  public void setup() {
    arr = new int[] {5, 3, 2, 1, 5};
  }

  @Test
  public void test() {
    HeapSort.sort(arr);
    assertArrayEquals(new int[] {1, 2, 3, 5, 5}, arr);
  }
}
